GitUI/CommandsDialogs/FormBrowse.cs

```diff
@@ -1803,6 +1803,11 @@ private void PopulateRecentRepositoriesMenu(ToolStripDropDownItem container)
 
             void AddRecentRepositories(Repository repo, string caption)
             {
+                if (AppSettings.RecentReposShowBranchName)
+                {
+                    caption = $"[{GitModule.GetSelectedBranchFast(repo.Path)}] {caption} ";
```

Comment:
I suggest the name is put after the path

Reply:
Basically, this information must be updated -- both for the active repo and for those of other GE instances. Is it?
How expensive is `GitModule.GetSelectedBranchFast`?

Reply:
Will do

Reply:
> 
> 
> Basically, this information must be updated -- both for the active repo and for those of other GE instances. Is it?
> How expensive is `GitModule.GetSelectedBranchFast`?

I implemented a very simple loop and called it 1000 times on my Ryzen 2500U with an SSD. Profiled it with DotTrace and seems like it takes ~200 microseconds for one call. I am no expert in profiling though...

Reply:
The call parses the file system to find the head. So expect it to require hundreds of ms in the worst case. Probably acceptable.